src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplication.cs

```diff
@@ -0,0 +1,209 @@
+{
+    /// <summary>
+    /// The class for supporting multi-components based application model.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class ComponentBasedApplication : Application
+    {
+        private const string LogTag = "Tizen.Applications.ComponentBasedApplicationBase";
+        private Dictionary<Type, ComponentFactoryBase> _componentFactories = new Dictionary<Type, ComponentFactoryBase>();
+        private Interop.CBApplication.CBAppLifecycleCallbacks _callbacks;
+        private bool _disposedValue = false;
+
+        /// <summary>
+        /// Initializes the ComponentBasedApplicationBase class.
+        /// </summary>
+        /// <param name="typeInfo">The component type information.
+        /// The key should be a class type of BaseComponent subclass.
+        /// The value should be a component id which is declared in tizen-manifest.xml.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public ComponentBasedApplication(IDictionary<Type, string> typeInfo)
+        {
+            _callbacks.OnInit = new Interop.CBApplication.CBAppInitCallback(OnInitNative);
+            _callbacks.OnFinished = new Interop.CBApplication.CBAppFiniCallback(OnFinishedNative);
+            _callbacks.OnRun = new Interop.CBApplication.CBAppRunCallback(OnRunNative);
+            _callbacks.OnExit = new Interop.CBApplication.CBAppExitCallback(OnExitNative);
+            _callbacks.OnCreate = new Interop.CBApplication.CBAppCreateCallback(OnCreateNative);
+            _callbacks.OnTerminate = new Interop.CBApplication.CBAppTerminateCallback(OnTerminateNative);
+
+            foreach (var component in typeInfo)
+            {
+                RegisterComponent(component.Key, component.Value);
+            }
+        }
+
+        /// <summary>
+        /// Registers a component.
+        /// </summary>
+        /// <param name="compType">Class type</param>
+        /// <param name="compId">Component ID</param>
+        /// <exception cref="InvalidOperationException">Thrown when component type is already added.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void RegisterComponent(Type compType, string compId)
+        {
+            if (_componentFactories.ContainsKey(compType))
+            {
+                throw new InvalidOperationException("Already exist type");
+            }
+
+            if (typeof(FrameComponent).IsAssignableFrom(compType))
+            {
+                Log.Info(LogTag, "Add frame component");
+                _componentFactories.Add(compType, new FrameFactory(compType, compId, this));
+            }
+            else if (typeof(ServiceComponent).IsAssignableFrom(compType))
+            {
+                Log.Info(LogTag, "Add service component");
+                _componentFactories.Add(compType, new ServiceFactory(compType, compId, this));
+            }
+            else if (typeof(BaseComponent).IsAssignableFrom(compType))
```

Comment:
I don't understadn why do you allow create of BaseComponent?
I think, Component should be one of FrameComponent or ServiceComponent
because ComponentType is only exist Frame or Service
``` c#
    public enum ComponentType
    {
        /// <summary>
        /// The frame component.
        /// </summary>
        Frame = Interop.CBApplication.NativeComponentType.Frame,

        /// <summary>
        /// The service component.
        /// </summary>
        Service = Interop.CBApplication.NativeComponentType.Service
    }
```

Reply:
I updated codes. Now, component-based applications cannot create BaseComponent instance.
Only sub-class of FrameComponent and ServiceComponent can be registered by component-based applications.